pythonPackages.twisted: 20.3.0 -> 21.7.0

exarkun opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

* Bring over a year of Twisted improvements into Nixpkgs
* Provide better type hints for Twisted APIs
* Provide minimum version required to run latest development version of GridSync (not in nixpkgs yet) 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

exarkun committed:
pythonPackages.twisted: 20.3.0 -> 21.7.0

risicle commented:
This breaks `thrift` because it tries to build its own python 2.7 version of twisted, and:

```
builder for '/nix/store/a2knnv0ma7nj5sairyn3y5n32apyinlk-python2.7-Twisted-21.7.0.drv' failed with exit code 1; last 10 log lines:
  configuring
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 10, in <module>
      import pathlib
  ImportError: No module named pathlib
```

Seems 20.3.0 was the last 2.7-supporting release.

risicle commented:
Also breaks `python3Packages.txtorcon`

risicle commented:
`python3Packages.klein` was already broken but now the tests fail earlier with:

```
builder for '/nix/store/kh4q0m6gwq0p0bvb3wmrn4hn6v5gds92-python3.9-klein-17.10.0.drv' failed with exit code 1; last 10 log lines:
      raise excValue.with_traceback(excTraceback)
    File "/nix/store/ivyddgcrkm01y46s9wg1mrhgmshb87hv-python3.9-klein-17.10.0/lib/python3.9/site-packages/klein/test/test_resource.py", line 13, in <module>
      from twisted.python.compat import _PY3, unicode
  builtins.ImportError: cannot import name '_PY3' from 'twisted.python.compat' (unknown location)

  klein.test.test_resource
  -------------------------------------------------------------------------------
  Ran 27 tests in 0.044s

  FAILED (errors=2, successes=25)
```

exarkun commented:
> Also breaks `python3Packages.txtorcon`, tests failing with e.g.:
> 
> ```
> >           result.addSkip(self, skipReason)
> E           TypeError: Skipped expected string as 'msg' parameter, got 'bool' instead.
> E           Perhaps you meant to use a mark?
> 
> 
> /nix/store/73djff0qryqiv6djqysjwylkqwlvmsp3-python3.9-Twisted-21.7.0/lib/python3.9/site-packages/twisted/trial/_synctest.py:1024: TypeError
> ```

Upstream is working on a fix for this - https://github.com/meejah/txtorcon/pull/351
But also they told me that txtorcon test suite is not typically run with pytest but with trial.

So is the right thing to do to wait for a txtorcon fix to be released, then upgrade txtorcon, then upgrade twisted?

exarkun commented:
> This breaks `thrift` because that tries to build its own python 2.7 version of twisted, and:
> 
> ```
> builder for '/nix/store/a2knnv0ma7nj5sairyn3y5n32apyinlk-python2.7-Twisted-21.7.0.drv' failed with exit code 1; last 10 log lines:
>   configuring
>   no configure script, doing nothing
>   building
>   Executing setuptoolsBuildPhase
>   Traceback (most recent call last):
>     File "nix_run_setup", line 8, in <module>
>       exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
>     File "setup.py", line 10, in <module>
>       import pathlib
>   ImportError: No module named pathlib
> ```
> 
> Seems 20.3.0 was the last 2.7-supporting release.

What is generally done about this in nixpkgs?  Is there a way to diverge the version in python2Packages vs python3Packages?

exarkun committed:
pythonPackages.tubes: init at 0.2.0

exarkun committed:
pythonPackages.klein: 17.10.0 -> 21.8.0

exarkun commented:
> `python3Packages.klein` was already broken but now the tests fail earlier with:
> 
> ```
> builder for '/nix/store/kh4q0m6gwq0p0bvb3wmrn4hn6v5gds92-python3.9-klein-17.10.0.drv' failed with exit code 1; last 10 log lines:
>       raise excValue.with_traceback(excTraceback)
>     File "/nix/store/ivyddgcrkm01y46s9wg1mrhgmshb87hv-python3.9-klein-17.10.0/lib/python3.9/site-packages/klein/test/test_resource.py", line 13, in <module>
>       from twisted.python.compat import _PY3, unicode
>   builtins.ImportError: cannot import name '_PY3' from 'twisted.python.compat' (unknown location)
> 
>   klein.test.test_resource
>   -------------------------------------------------------------------------------
>   Ran 27 tests in 0.044s
> 
>   FAILED (errors=2, successes=25)
> ```

Okay.  I have a branch that upgrades klein, too, to deal with this.  It needs https://github.com/NixOS/nixpkgs/pull/132944 first though.

jonringer commented:
> This breaks `thrift` because that tries to build its own python 2.7 version of twisted, and:

How has an internet-facing apache project not migrated off of python2 after 18months of it being EoL?

SuperSandro2000 reviewed (commented):
You want to add yourself as a maintainer?

SuperSandro2000 commented:
> What is generally done about this in nixpkgs? Is there a way to diverge the version in python2Packages vs python3Packages?

python2.7 is EOL. If it isn't used by nixops then we can safely disable it for older versions.

> Upstream is working on a fix for this - meejah/txtorcon#351
But also they told me that txtorcon test suite is not typically run with pytest but with trial.

So is the right thing to do to wait for a txtorcon fix to be released, then upgrade txtorcon, then upgrade twisted?

Or just mark it broken for now or pull in the patch. Python packages is so big we can't wait for everything and Twisted seems to be more important.

risicle commented:
`txtorcon` has reverse dependencies, such as `magic-wormhole` which seems to get quite a lot of use.

An alternative is to apply https://github.com/meejah/txtorcon/pull/351 (that would be https://github.com/meejah/txtorcon/commit/94ca067373e10bd1af3a27b9f4a5cf0dc958d803) to `txtorcon` until they come up with a release. Note how that _looks_ like a big patch until you notice most of it is stuff to do with their own CI and we could probably use `fetchpatch` to filter it down to just the `test/test_util.py` and `txtorcon/torinfo.py` changes.

mweinelt commented:
>  But also they told me that txtorcon test suite is not typically run with pytest but with trial.

Then let's migrate that to trial.

exarkun committed:
pythonPackages.txtorcon: run tests with trial and disable for <py3.5

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 132900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages marked as broken and skipped:
</summary>
<ul>
<li>pypi2nix</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pymatgen-lammps</li>
<li>python38Packages.skein</li>
<li>python39Packages.optuna</li>
<li>python39Packages.pymatgen-lammps</li>
<li>python39Packages.skein</li>
<li>thrift-0_10</li>
<li>tribler</li>
</ul>
</details>
<details>

  <summary>51 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>apache-airflow: log was empty</li>
<li>aws-sam-cli: log was empty</li>
<li>cq-editor: log was empty</li>
<li>home-assistant: log was empty</li>
<li>python38Packages.apache-airflow: log was empty</li>
<li>python38Packages.asdf: log was empty</li>
<li>python38Packages.carbon: log was empty</li>
<li>python38Packages.graphite-web: log was empty</li>
<li>python38Packages.ipydatawidgets: log was empty</li>
<li>python38Packages.jupyterlab: log was empty</li>
<li>python38Packages.jupyterlab-git: log was empty</li>
<li>python38Packages.jupyterlab_server: log was empty</li>
<li>python38Packages.jupytext: log was empty</li>
<li>python38Packages.klein: log was empty</li>
<li>python38Packages.nbdime: log was empty</li>
<li>python38Packages.openapi-core: log was empty</li>
<li>python38Packages.pathy: log was empty</li>
<li>python38Packages.robotframework-selenium2library: log was empty</li>
<li>python38Packages.robotframework-seleniumlibrary: log was empty</li>
<li>python38Packages.robotframework-tools: log was empty</li>
<li>python38Packages.spacy: log was empty</li>
<li>python38Packages.spacy-transformers: log was empty</li>
<li>python38Packages.sunpy: log was empty</li>
<li>python38Packages.textacy: log was empty</li>
<li>python38Packages.tokenizers: log was empty</li>
<li>python38Packages.transformers: log was empty</li>
<li>python39Packages.apache-airflow: log was empty</li>
<li>python39Packages.asdf: log was empty</li>
<li>python39Packages.carbon: log was empty</li>
<li>python39Packages.google-cloud-bigquery: log was empty</li>
<li>python39Packages.graphite-web: log was empty</li>
<li>python39Packages.ipydatawidgets: log was empty</li>
<li>python39Packages.jupyterlab: log was empty</li>
<li>python39Packages.jupyterlab-git: log was empty</li>
<li>python39Packages.jupyterlab_server: log was empty</li>
<li>python39Packages.jupytext: log was empty</li>
<li>python39Packages.klein: log was empty</li>
<li>python39Packages.nbdime: log was empty</li>
<li>python39Packages.openapi-core: log was empty</li>
<li>python39Packages.pathy: log was empty</li>
<li>python39Packages.robotframework-selenium2library: log was empty</li>
<li>python39Packages.robotframework-seleniumlibrary: log was empty</li>
<li>python39Packages.robotframework-tools: log was empty</li>
<li>python39Packages.spacy: log was empty</li>
<li>python39Packages.spacy-transformers: log was empty</li>
<li>python39Packages.sunpy: log was empty</li>
<li>python39Packages.textacy: log was empty</li>
<li>python39Packages.tokenizers: log was empty</li>
<li>python39Packages.transformers: log was empty</li>
<li>theharvester: log was empty</li>
<li>tts: log was empty</li>
  </ul>
</details>
<details>

  <summary>31 packages failed to build and are new build failures:</summary>
  <ul>
<li>arrow-cpp: log was empty</li>
<li>deluge-1_x: log was empty</li>
<li>gnome-keysign: log was empty</li>
<li>magic-wormhole: log was empty</li>
<li>(python39Packages.magic-wormhole): log was empty</li>
<li>python38Packages.catboost: log was empty</li>
<li>python38Packages.datasets: log was empty</li>
<li>python38Packages.google-cloud-bigquery: log was empty</li>
<li>python38Packages.intake: log was empty</li>
<li>python38Packages.magic-wormhole: log was empty</li>
<li>python38Packages.pyarrow: log was empty</li>
<li>python38Packages.qiskit: log was empty</li>
<li>python38Packages.qiskit-aqua: log was empty</li>
<li>python38Packages.snowflake-connector-python: log was empty</li>
<li>python38Packages.snowflake-sqlalchemy: log was empty</li>
<li>python38Packages.txtorcon: log was empty</li>
<li>python39Packages.batchspawner: log was empty</li>
<li>python39Packages.datasets: log was empty</li>
<li>python39Packages.dockerspawner: log was empty</li>
<li>python39Packages.intake: log was empty</li>
<li>python39Packages.jupyterhub: log was empty</li>
<li>python39Packages.jupyterhub-ldapauthenticator: log was empty</li>
<li>python39Packages.jupyterhub-systemdspawner: log was empty</li>
<li>python39Packages.jupyterhub-tmpauthenticator: log was empty</li>
<li>python39Packages.oauthenticator: log was empty</li>
<li>python39Packages.pyarrow: log was empty</li>
<li>python39Packages.snowflake-connector-python: log was empty</li>
<li>python39Packages.snowflake-sqlalchemy: log was empty</li>
<li>python39Packages.txtorcon: log was empty</li>
<li>tahoe-lafs: log was empty</li>
<li>thrift: log was empty</li>
  </ul>
</details>
<details>
<summary>
375 packages built:
</summary>
<ul>
<li>alot (python39Packages.alot)</li>
<li>anki</li>
<li>ankisyncd</li>
<li>ansible (ansible_2_11 ,python39Packages.ansible-core)</li>
<li>ansible-lint (python39Packages.ansible-lint)</li>
<li>ansible_2_10 (python39Packages.ansible-base)</li>
<li>apio</li>
<li>arion</li>
<li>awscli2</li>
<li>blocksat-cli (python39Packages.blocksat-cli)</li>
<li>borgmatic</li>
<li>buildbot (python39Packages.buildbot)</li>
<li>buildbot-full (python39Packages.buildbot-full)</li>
<li>buildbot-ui (python39Packages.buildbot-ui)</li>
<li>buildbot-worker (python39Packages.buildbot-worker)</li>
<li>bump2version</li>
<li>check-esxi-hardware</li>
<li>cloud-custodian</li>
<li>cloud-init</li>
<li>cyclonedx-python</li>
<li>deeptools</li>
<li>deluge (deluge-2_x)</li>
<li>docker-compose</li>
<li>dt-schema</li>
<li>expliot</li>
<li>flexget</li>
<li>giara</li>
<li>gns3-gui</li>
<li>gns3-server</li>
<li>grab-site</li>
<li>haskellPackages.nvfetcher</li>
<li>hydrus</li>
<li>ihaskell</li>
<li>inkcut</li>
<li>inkscape-extensions.inkcut</li>
<li>jupyter</li>
<li>matrix-commander</li>
<li>matrix-synapse</li>
<li>matrix-synapse-plugins.matrix-synapse-ldap3</li>
<li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
<li>matrix-synapse-plugins.matrix-synapse-pam</li>
<li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth</li>
<li>mirage-im</li>
<li>mnemosyne</li>
<li>nbstripout</li>
<li>nvchecker (python39Packages.nvchecker)</li>
<li>nvfetcher</li>
<li>opsdroid</li>
<li>pantalaimon</li>
<li>powerdns-admin</li>
<li>privacyidea</li>
<li>python38Packages.APScheduler</li>
<li>python38Packages.Nikola</li>
<li>python38Packages.Pweave</li>
<li>python38Packages.alot</li>
<li>python38Packages.altair</li>
<li>python38Packages.ansible-base</li>
<li>python38Packages.ansible-core</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.ansible-lint</li>
<li>python38Packages.apispec</li>
<li>python38Packages.autobahn</li>
<li>python38Packages.aws-sam-translator</li>
<li>python38Packages.batchspawner</li>
<li>python38Packages.blocksat-cli</li>
<li>python38Packages.bravado-core</li>
<li>python38Packages.buildbot</li>
<li>python38Packages.buildbot-full</li>
<li>python38Packages.buildbot-ui</li>
<li>python38Packages.buildbot-worker</li>
<li>python38Packages.cassandra-driver</li>
<li>python38Packages.castepxbin</li>
<li>python38Packages.celery</li>
<li>python38Packages.cfn-lint</li>
<li>python38Packages.channels</li>
<li>python38Packages.channels-redis</li>
<li>python38Packages.chart-studio</li>
<li>python38Packages.connexion</li>
<li>python38Packages.cufflinks</li>
<li>python38Packages.cve-bin-tool</li>
<li>python38Packages.daphne</li>
<li>python38Packages.dash</li>
<li>python38Packages.datashader</li>
<li>python38Packages.django-raster</li>
<li>python38Packages.django-simple-captcha</li>
<li>python38Packages.djmail</li>
<li>python38Packages.dockerspawner</li>
<li>python38Packages.flask-appbuilder</li>
<li>python38Packages.flask-restplus</li>
<li>python38Packages.flask-restx</li>
<li>python38Packages.flower</li>
<li>python38Packages.foolscap</li>
<li>python38Packages.gensim</li>
<li>python38Packages.geoip2</li>
<li>python38Packages.graphite_api</li>
<li>python38Packages.holoviews</li>
<li>python38Packages.hvplot</li>
<li>python38Packages.influxgraph</li>
<li>python38Packages.ipympl</li>
<li>python38Packages.ipyvue</li>
<li>python38Packages.ipyvuetify</li>
<li>python38Packages.ipywidgets</li>
<li>python38Packages.jsonmerge</li>
<li>python38Packages.jsonschema</li>
<li>python38Packages.jupyter</li>
<li>python38Packages.jupyter-sphinx</li>
<li>python38Packages.jupyter-telemetry</li>
<li>python38Packages.jupyter_server</li>
<li>python38Packages.jupyterhub</li>
<li>python38Packages.jupyterhub-ldapauthenticator</li>
<li>python38Packages.jupyterhub-systemdspawner</li>
<li>python38Packages.jupyterhub-tmpauthenticator</li>
<li>python38Packages.jupyterlab_launcher</li>
<li>python38Packages.kmapper</li>
<li>python38Packages.labgrid</li>
<li>python38Packages.ldaptor</li>
<li>python38Packages.lightparam</li>
<li>python38Packages.livelossplot</li>
<li>python38Packages.magic-wormhole-mailbox-server</li>
<li>python38Packages.magic-wormhole-transit-relay</li>
<li>python38Packages.matrix-nio</li>
<li>python38Packages.mocket</li>
<li>python38Packages.modeled</li>
<li>python38Packages.moretools</li>
<li>python38Packages.moto</li>
<li>python38Packages.nbclassic</li>
<li>python38Packages.nbclient</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbconvert</li>
<li>python38Packages.nbformat</li>
<li>python38Packages.nbmerge</li>
<li>python38Packages.nbsmoke</li>
<li>python38Packages.nbsphinx</li>
<li>python38Packages.nbval</li>
<li>python38Packages.nix-kernel</li>
<li>python38Packages.notebook</li>
<li>python38Packages.notedown</li>
<li>python38Packages.nvchecker</li>
<li>python38Packages.oauthenticator</li>
<li>python38Packages.onnx</li>
<li>python38Packages.openapi-schema-validator</li>
<li>python38Packages.openapi-spec-validator</li>
<li>python38Packages.openerz-api</li>
<li>python38Packages.papermill</li>
<li>python38Packages.phik</li>
<li>python38Packages.pika</li>
<li>python38Packages.pika-pool</li>
<li>python38Packages.plotly</li>
<li>python38Packages.pook</li>
<li>python38Packages.prance</li>
<li>python38Packages.praw</li>
<li>python38Packages.prawcore</li>
<li>python38Packages.pyee</li>
<li>python38Packages.pykwalify</li>
<li>python38Packages.pymatgen</li>
<li>python38Packages.pymodbus</li>
<li>python38Packages.pyppeteer</li>
<li>python38Packages.python-telegram-bot</li>
<li>python38Packages.pyutil</li>
<li>python38Packages.pywbem</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.qreactor</li>
<li>python38Packages.rachiopy</li>
<li>python38Packages.robomachine</li>
<li>python38Packages.robotframework</li>
<li>python38Packages.robotframework-databaselibrary</li>
<li>python38Packages.robotframework-requests</li>
<li>python38Packages.robotframework-sshlibrary</li>
<li>python38Packages.robotstatuschecker</li>
<li>python38Packages.robotsuite</li>
<li>python38Packages.scikit-tda</li>
<li>python38Packages.scrapy</li>
<li>python38Packages.scrapy-deltafetch</li>
<li>python38Packages.scrapy-fake-useragent</li>
<li>python38Packages.scrapy-splash</li>
<li>python38Packages.setuptoolsTrial</li>
<li>python38Packages.smart-open</li>
<li>python38Packages.sopel</li>
<li>python38Packages.sphinxcontrib-openapi</li>
<li>python38Packages.spyder</li>
<li>python38Packages.structlog</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sumo</li>
<li>python38Packages.swagger-spec-validator</li>
<li>python38Packages.synergy</li>
<li>python38Packages.tensorboardx</li>
<li>python38Packages.testfixtures</li>
<li>python38Packages.towncrier</li>
<li>python38Packages.treq</li>
<li>python38Packages.ttp</li>
<li>python38Packages.twisted</li>
<li>python38Packages.txaio</li>
<li>python38Packages.txamqp</li>
<li>python38Packages.txdbus</li>
<li>python38Packages.txgithub</li>
<li>python38Packages.txredisapi</li>
<li>python38Packages.txrequests</li>
<li>python38Packages.vega</li>
<li>python38Packages.vncdo</li>
<li>python38Packages.warlock</li>
<li>python38Packages.webthing</li>
<li>python38Packages.west</li>
<li>python38Packages.widgetsnbextension</li>
<li>python38Packages.zetup</li>
<li>python38Packages.zfec</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.APScheduler</li>
<li>python39Packages.Nikola</li>
<li>python39Packages.Pweave</li>
<li>python39Packages.altair</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.apispec</li>
<li>python39Packages.autobahn</li>
<li>python39Packages.aws-sam-translator</li>
<li>python39Packages.bravado-core</li>
<li>python39Packages.cassandra-driver</li>
<li>python39Packages.castepxbin</li>
<li>python39Packages.catboost</li>
<li>python39Packages.celery</li>
<li>python39Packages.cfn-lint</li>
<li>python39Packages.channels</li>
<li>python39Packages.channels-redis</li>
<li>python39Packages.chart-studio</li>
<li>python39Packages.connexion</li>
<li>python39Packages.cufflinks</li>
<li>python39Packages.cve-bin-tool</li>
<li>python39Packages.daphne</li>
<li>python39Packages.dash</li>
<li>python39Packages.datashader</li>
<li>python39Packages.django-raster</li>
<li>python39Packages.django-simple-captcha</li>
<li>python39Packages.djmail</li>
<li>python39Packages.flask-appbuilder</li>
<li>python39Packages.flask-restplus</li>
<li>python39Packages.flask-restx</li>
<li>python39Packages.flower</li>
<li>python39Packages.foolscap</li>
<li>python39Packages.gensim</li>
<li>python39Packages.geoip2</li>
<li>python39Packages.graphite_api</li>
<li>python39Packages.holoviews</li>
<li>python39Packages.hvplot</li>
<li>python39Packages.influxgraph</li>
<li>python39Packages.ipympl</li>
<li>python39Packages.ipyvue</li>
<li>python39Packages.ipyvuetify</li>
<li>python39Packages.ipywidgets</li>
<li>python39Packages.jsonmerge</li>
<li>python39Packages.jsonschema</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.jupyter-sphinx</li>
<li>python39Packages.jupyter-telemetry</li>
<li>python39Packages.jupyter_server</li>
<li>python39Packages.jupyterlab_launcher</li>
<li>python39Packages.kmapper</li>
<li>python39Packages.labgrid</li>
<li>python39Packages.ldaptor</li>
<li>python39Packages.lightparam</li>
<li>python39Packages.livelossplot</li>
<li>python39Packages.magic-wormhole-mailbox-server</li>
<li>python39Packages.magic-wormhole-transit-relay</li>
<li>python39Packages.matrix-nio</li>
<li>python39Packages.mocket</li>
<li>python39Packages.modeled</li>
<li>python39Packages.moretools</li>
<li>python39Packages.moto</li>
<li>python39Packages.nbclassic</li>
<li>python39Packages.nbclient</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbconvert</li>
<li>python39Packages.nbformat</li>
<li>python39Packages.nbmerge</li>
<li>python39Packages.nbsmoke</li>
<li>python39Packages.nbsphinx</li>
<li>python39Packages.nbval</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.notebook</li>
<li>python39Packages.notedown</li>
<li>python39Packages.onnx</li>
<li>python39Packages.openapi-schema-validator</li>
<li>python39Packages.openapi-spec-validator</li>
<li>python39Packages.openerz-api</li>
<li>python39Packages.papermill</li>
<li>python39Packages.phik</li>
<li>python39Packages.pika</li>
<li>python39Packages.pika-pool</li>
<li>python39Packages.plotly</li>
<li>python39Packages.pook</li>
<li>python39Packages.prance</li>
<li>python39Packages.praw</li>
<li>python39Packages.prawcore</li>
<li>python39Packages.pyee</li>
<li>python39Packages.pykwalify</li>
<li>python39Packages.pymatgen</li>
<li>python39Packages.pymodbus</li>
<li>python39Packages.pyppeteer</li>
<li>python39Packages.python-telegram-bot</li>
<li>python39Packages.pyutil</li>
<li>python39Packages.pywbem</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.qreactor</li>
<li>python39Packages.rachiopy</li>
<li>python39Packages.robomachine</li>
<li>python39Packages.robotframework</li>
<li>python39Packages.robotframework-databaselibrary</li>
<li>python39Packages.robotframework-requests</li>
<li>python39Packages.robotframework-sshlibrary</li>
<li>python39Packages.robotstatuschecker</li>
<li>python39Packages.robotsuite</li>
<li>python39Packages.scikit-tda</li>
<li>python39Packages.scrapy</li>
<li>python39Packages.scrapy-deltafetch</li>
<li>python39Packages.scrapy-fake-useragent</li>
<li>python39Packages.scrapy-splash</li>
<li>python39Packages.setuptoolsTrial</li>
<li>python39Packages.smart-open</li>
<li>python39Packages.sopel</li>
<li>python39Packages.sphinxcontrib-openapi</li>
<li>python39Packages.structlog</li>
<li>python39Packages.stytra</li>
<li>python39Packages.sumo</li>
<li>python39Packages.swagger-spec-validator</li>
<li>python39Packages.synergy</li>
<li>python39Packages.tensorboardx</li>
<li>python39Packages.testfixtures</li>
<li>python39Packages.towncrier</li>
<li>python39Packages.treq</li>
<li>ttp (python39Packages.ttp)</li>
<li>python39Packages.twisted</li>
<li>python39Packages.txaio</li>
<li>python39Packages.txamqp</li>
<li>python39Packages.txdbus</li>
<li>python39Packages.txgithub</li>
<li>python39Packages.txredisapi</li>
<li>python39Packages.txrequests</li>
<li>python39Packages.vega</li>
<li>vncdo (python39Packages.vncdo)</li>
<li>python39Packages.warlock</li>
<li>python39Packages.webthing</li>
<li>python39Packages.west</li>
<li>python39Packages.widgetsnbextension</li>
<li>python39Packages.zetup</li>
<li>python39Packages.zfec</li>
<li>python39Packages.zigpy-znp</li>
<li>qmk</li>
<li>rmview</li>
<li>sage</li>
<li>sageWithDoc</li>
<li>snakemake</li>
<li>sourcehut.buildsrht</li>
<li>sourcehut.coresrht</li>
<li>sourcehut.dispatchsrht</li>
<li>sourcehut.gitsrht</li>
<li>sourcehut.hgsrht</li>
<li>sourcehut.hubsrht</li>
<li>sourcehut.listssrht</li>
<li>sourcehut.mansrht</li>
<li>sourcehut.metasrht</li>
<li>sourcehut.pastesrht</li>
<li>sourcehut.todosrht</li>
<li>streamlit</li>
<li>syncplay</li>
<li>tinyprog</li>
<li>urlwatch</li>
<li>vorta</li>
<li>weechatScripts.weechat-matrix</li>
<li>whipper</li>
</ul>
</details>

exarkun committed:
thrift: Move from Python 2 to Python 3

The Twisted dependency now requires Python 3.
Also, Python 2 is long-since EOL.

exarkun committed:
maintainers: add exarkun

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 132900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
17 packages marked as broken and skipped:
</summary>
<ul>
<li>pypi2nix</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pathy</li>
<li>python38Packages.pymatgen-lammps</li>
<li>python38Packages.skein</li>
<li>python38Packages.spacy</li>
<li>python38Packages.spacy-transformers</li>
<li>python38Packages.textacy</li>
<li>python39Packages.optuna</li>
<li>python39Packages.pathy</li>
<li>python39Packages.pymatgen-lammps</li>
<li>python39Packages.skein</li>
<li>python39Packages.spacy</li>
<li>python39Packages.spacy-transformers</li>
<li>python39Packages.textacy</li>
<li>thrift-0_10</li>
<li>tribler</li>
</ul>
</details>
<details>

  <summary>27 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>apache-airflow: log was empty</li>
<li>cq-editor: log was empty</li>
<li>paperless-ng: log was empty</li>
<li>python38Packages.apache-airflow: log was empty</li>
<li>python38Packages.carbon: log was empty</li>
<li>python38Packages.datashader: log was empty</li>
<li>python38Packages.graphite-web: log was empty</li>
<li>python38Packages.ipydatawidgets: log was empty</li>
<li>python38Packages.jupyterlab-git: log was empty</li>
<li>python38Packages.nbdime: log was empty</li>
<li>python38Packages.robotframework-selenium2library: log was empty</li>
<li>python38Packages.robotframework-seleniumlibrary: log was empty</li>
<li>python38Packages.robotframework-tools: log was empty</li>
<li>python38Packages.tokenizers: log was empty</li>
<li>python38Packages.transformers: log was empty</li>
<li>python39Packages.apache-airflow: log was empty</li>
<li>python39Packages.carbon: log was empty</li>
<li>python39Packages.graphite-web: log was empty</li>
<li>python39Packages.ipydatawidgets: log was empty</li>
<li>python39Packages.jupyterlab-git: log was empty</li>
<li>python39Packages.nbdime: log was empty</li>
<li>python39Packages.robotframework-selenium2library: log was empty</li>
<li>python39Packages.robotframework-seleniumlibrary: log was empty</li>
<li>python39Packages.robotframework-tools: log was empty</li>
<li>python39Packages.tokenizers: log was empty</li>
<li>python39Packages.transformers: log was empty</li>
<li>theharvester: log was empty</li>
  </ul>
</details>
<details>
<summary>
423 packages built:
</summary>
<ul>
<li>alot (python39Packages.alot)</li>
<li>anki</li>
<li>ankisyncd</li>
<li>ansible (ansible_2_11 ,python39Packages.ansible-core)</li>
<li>ansible-lint (python39Packages.ansible-lint)</li>
<li>ansible_2_10 (python39Packages.ansible-base)</li>
<li>apio</li>
<li>arion</li>
<li>arrow-cpp</li>
<li>aws-sam-cli</li>
<li>awscli2</li>
<li>blocksat-cli (python39Packages.blocksat-cli)</li>
<li>borgmatic</li>
<li>buildbot (python39Packages.buildbot)</li>
<li>buildbot-full (python39Packages.buildbot-full)</li>
<li>buildbot-ui (python39Packages.buildbot-ui)</li>
<li>buildbot-worker (python39Packages.buildbot-worker)</li>
<li>bump2version</li>
<li>check-esxi-hardware</li>
<li>cloud-custodian</li>
<li>cloud-init</li>
<li>cyclonedx-python</li>
<li>deeptools</li>
<li>deluge (deluge-2_x)</li>
<li>docker-compose</li>
<li>dt-schema</li>
<li>expliot</li>
<li>flexget</li>
<li>giara</li>
<li>gnome-keysign</li>
<li>gns3-gui</li>
<li>gns3-server</li>
<li>grab-site</li>
<li>haskellPackages.nvfetcher</li>
<li>home-assistant</li>
<li>hydrus</li>
<li>ihaskell</li>
<li>inkcut</li>
<li>inkscape-extensions.inkcut</li>
<li>jupyter</li>
<li>magic-wormhole (python39Packages.magic-wormhole)</li>
<li>matrix-commander</li>
<li>matrix-synapse</li>
<li>matrix-synapse-plugins.matrix-synapse-ldap3</li>
<li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
<li>matrix-synapse-plugins.matrix-synapse-pam</li>
<li>matrix-synapse-plugins.matrix-synapse-shared-secret-auth</li>
<li>mirage-im</li>
<li>mnemosyne</li>
<li>nbstripout</li>
<li>nvchecker (python39Packages.nvchecker)</li>
<li>nvfetcher</li>
<li>opsdroid</li>
<li>pantalaimon</li>
<li>powerdns-admin</li>
<li>privacyidea</li>
<li>python38Packages.APScheduler</li>
<li>python38Packages.Nikola</li>
<li>python38Packages.Pweave</li>
<li>python38Packages.alot</li>
<li>python38Packages.altair</li>
<li>python38Packages.ansible-base</li>
<li>python38Packages.ansible-core</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.ansible-lint</li>
<li>python38Packages.apispec</li>
<li>python38Packages.asdf</li>
<li>python38Packages.autobahn</li>
<li>python38Packages.aws-sam-translator</li>
<li>python38Packages.batchspawner</li>
<li>python38Packages.blocksat-cli</li>
<li>python38Packages.bravado-core</li>
<li>python38Packages.buildbot</li>
<li>python38Packages.buildbot-full</li>
<li>python38Packages.buildbot-ui</li>
<li>python38Packages.buildbot-worker</li>
<li>python38Packages.cassandra-driver</li>
<li>python38Packages.castepxbin</li>
<li>python38Packages.catboost</li>
<li>python38Packages.celery</li>
<li>python38Packages.cfn-lint</li>
<li>python38Packages.channels</li>
<li>python38Packages.channels-redis</li>
<li>python38Packages.chart-studio</li>
<li>python38Packages.connexion</li>
<li>python38Packages.cufflinks</li>
<li>python38Packages.cve-bin-tool</li>
<li>python38Packages.daphne</li>
<li>python38Packages.dash</li>
<li>python38Packages.datasets</li>
<li>python38Packages.django-raster</li>
<li>python38Packages.django-simple-captcha</li>
<li>python38Packages.djmail</li>
<li>python38Packages.dockerspawner</li>
<li>python38Packages.flask-appbuilder</li>
<li>python38Packages.flask-restplus</li>
<li>python38Packages.flask-restx</li>
<li>python38Packages.flower</li>
<li>python38Packages.foolscap</li>
<li>python38Packages.gensim</li>
<li>python38Packages.geoip2</li>
<li>python38Packages.google-cloud-bigquery</li>
<li>python38Packages.graphite_api</li>
<li>python38Packages.holoviews</li>
<li>python38Packages.hvplot</li>
<li>python38Packages.influxgraph</li>
<li>python38Packages.intake</li>
<li>python38Packages.ipympl</li>
<li>python38Packages.ipyvue</li>
<li>python38Packages.ipyvuetify</li>
<li>python38Packages.ipywidgets</li>
<li>python38Packages.jsonmerge</li>
<li>python38Packages.jsonschema</li>
<li>python38Packages.jupyter</li>
<li>python38Packages.jupyter-sphinx</li>
<li>python38Packages.jupyter-telemetry</li>
<li>python38Packages.jupyter_server</li>
<li>python38Packages.jupyterhub</li>
<li>python38Packages.jupyterhub-ldapauthenticator</li>
<li>python38Packages.jupyterhub-systemdspawner</li>
<li>python38Packages.jupyterhub-tmpauthenticator</li>
<li>python38Packages.jupyterlab</li>
<li>python38Packages.jupyterlab_launcher</li>
<li>python38Packages.jupyterlab_server</li>
<li>python38Packages.jupytext</li>
<li>python38Packages.klein</li>
<li>python38Packages.kmapper</li>
<li>python38Packages.labgrid</li>
<li>python38Packages.ldaptor</li>
<li>python38Packages.lightparam</li>
<li>python38Packages.livelossplot</li>
<li>python38Packages.magic-wormhole</li>
<li>python38Packages.magic-wormhole-mailbox-server</li>
<li>python38Packages.magic-wormhole-transit-relay</li>
<li>python38Packages.matrix-nio</li>
<li>python38Packages.mocket</li>
<li>python38Packages.modeled</li>
<li>python38Packages.moretools</li>
<li>python38Packages.moto</li>
<li>python38Packages.nbclassic</li>
<li>python38Packages.nbclient</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbconvert</li>
<li>python38Packages.nbformat</li>
<li>python38Packages.nbmerge</li>
<li>python38Packages.nbsmoke</li>
<li>python38Packages.nbsphinx</li>
<li>python38Packages.nbval</li>
<li>python38Packages.nix-kernel</li>
<li>python38Packages.notebook</li>
<li>python38Packages.notedown</li>
<li>python38Packages.nvchecker</li>
<li>python38Packages.oauthenticator</li>
<li>python38Packages.onnx</li>
<li>python38Packages.openapi-core</li>
<li>python38Packages.openapi-schema-validator</li>
<li>python38Packages.openapi-spec-validator</li>
<li>python38Packages.openerz-api</li>
<li>python38Packages.papermill</li>
<li>python38Packages.phik</li>
<li>python38Packages.pika</li>
<li>python38Packages.pika-pool</li>
<li>python38Packages.plotly</li>
<li>python38Packages.pook</li>
<li>python38Packages.prance</li>
<li>python38Packages.praw</li>
<li>python38Packages.prawcore</li>
<li>python38Packages.pyarrow</li>
<li>python38Packages.pyee</li>
<li>python38Packages.pykwalify</li>
<li>python38Packages.pymatgen</li>
<li>python38Packages.pymodbus</li>
<li>python38Packages.pyppeteer</li>
<li>python38Packages.python-telegram-bot</li>
<li>python38Packages.pyutil</li>
<li>python38Packages.pywbem</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.qreactor</li>
<li>python38Packages.rachiopy</li>
<li>python38Packages.robomachine</li>
<li>python38Packages.robotframework</li>
<li>python38Packages.robotframework-databaselibrary</li>
<li>python38Packages.robotframework-requests</li>
<li>python38Packages.robotframework-sshlibrary</li>
<li>python38Packages.robotstatuschecker</li>
<li>python38Packages.robotsuite</li>
<li>python38Packages.scikit-tda</li>
<li>python38Packages.scrapy</li>
<li>python38Packages.scrapy-deltafetch</li>
<li>python38Packages.scrapy-fake-useragent</li>
<li>python38Packages.scrapy-splash</li>
<li>python38Packages.setuptoolsTrial</li>
<li>python38Packages.slack-sdk</li>
<li>python38Packages.smart-open</li>
<li>python38Packages.snowflake-connector-python</li>
<li>python38Packages.snowflake-sqlalchemy</li>
<li>python38Packages.sopel</li>
<li>python38Packages.sphinxcontrib-openapi</li>
<li>python38Packages.spyder</li>
<li>python38Packages.structlog</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sumo</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.swagger-spec-validator</li>
<li>python38Packages.synergy</li>
<li>python38Packages.tensorboardx</li>
<li>python38Packages.testfixtures</li>
<li>python38Packages.towncrier</li>
<li>python38Packages.treq</li>
<li>python38Packages.ttp</li>
<li>python38Packages.tubes</li>
<li>python38Packages.twisted</li>
<li>python38Packages.txaio</li>
<li>python38Packages.txamqp</li>
<li>python38Packages.txdbus</li>
<li>python38Packages.txgithub</li>
<li>python38Packages.txredisapi</li>
<li>python38Packages.txrequests</li>
<li>python38Packages.txtorcon</li>
<li>python38Packages.vega</li>
<li>python38Packages.vncdo</li>
<li>python38Packages.warlock</li>
<li>python38Packages.webthing</li>
<li>python38Packages.west</li>
<li>python38Packages.widgetsnbextension</li>
<li>python38Packages.zetup</li>
<li>python38Packages.zfec</li>
<li>python38Packages.zigpy-znp</li>
<li>python39Packages.APScheduler</li>
<li>python39Packages.Nikola</li>
<li>python39Packages.Pweave</li>
<li>python39Packages.altair</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.apispec</li>
<li>python39Packages.asdf</li>
<li>python39Packages.autobahn</li>
<li>python39Packages.aws-sam-translator</li>
<li>python39Packages.batchspawner</li>
<li>python39Packages.bravado-core</li>
<li>python39Packages.cassandra-driver</li>
<li>python39Packages.castepxbin</li>
<li>python39Packages.catboost</li>
<li>python39Packages.celery</li>
<li>python39Packages.cfn-lint</li>
<li>python39Packages.channels</li>
<li>python39Packages.channels-redis</li>
<li>python39Packages.chart-studio</li>
<li>python39Packages.connexion</li>
<li>python39Packages.cufflinks</li>
<li>python39Packages.cve-bin-tool</li>
<li>python39Packages.daphne</li>
<li>python39Packages.dash</li>
<li>python39Packages.datasets</li>
<li>python39Packages.datashader</li>
<li>python39Packages.django-raster</li>
<li>python39Packages.django-simple-captcha</li>
<li>python39Packages.djmail</li>
<li>python39Packages.dockerspawner</li>
<li>python39Packages.flask-appbuilder</li>
<li>python39Packages.flask-restplus</li>
<li>python39Packages.flask-restx</li>
<li>python39Packages.flower</li>
<li>python39Packages.foolscap</li>
<li>python39Packages.gensim</li>
<li>python39Packages.geoip2</li>
<li>python39Packages.google-cloud-bigquery</li>
<li>python39Packages.graphite_api</li>
<li>python39Packages.holoviews</li>
<li>python39Packages.hvplot</li>
<li>python39Packages.influxgraph</li>
<li>python39Packages.intake</li>
<li>python39Packages.ipympl</li>
<li>python39Packages.ipyvue</li>
<li>python39Packages.ipyvuetify</li>
<li>python39Packages.ipywidgets</li>
<li>python39Packages.jsonmerge</li>
<li>python39Packages.jsonschema</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.jupyter-sphinx</li>
<li>python39Packages.jupyter-telemetry</li>
<li>python39Packages.jupyter_server</li>
<li>python39Packages.jupyterhub</li>
<li>python39Packages.jupyterhub-ldapauthenticator</li>
<li>python39Packages.jupyterhub-systemdspawner</li>
<li>python39Packages.jupyterhub-tmpauthenticator</li>
<li>python39Packages.jupyterlab</li>
<li>python39Packages.jupyterlab_launcher</li>
<li>python39Packages.jupyterlab_server</li>
<li>python39Packages.jupytext</li>
<li>python39Packages.klein</li>
<li>python39Packages.kmapper</li>
<li>python39Packages.labgrid</li>
<li>python39Packages.ldaptor</li>
<li>python39Packages.lightparam</li>
<li>python39Packages.livelossplot</li>
<li>python39Packages.magic-wormhole-mailbox-server</li>
<li>python39Packages.magic-wormhole-transit-relay</li>
<li>python39Packages.matrix-nio</li>
<li>python39Packages.mocket</li>
<li>python39Packages.modeled</li>
<li>python39Packages.moretools</li>
<li>python39Packages.moto</li>
<li>python39Packages.nbclassic</li>
<li>python39Packages.nbclient</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbconvert</li>
<li>python39Packages.nbformat</li>
<li>python39Packages.nbmerge</li>
<li>python39Packages.nbsmoke</li>
<li>python39Packages.nbsphinx</li>
<li>python39Packages.nbval</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.notebook</li>
<li>python39Packages.notedown</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.onnx</li>
<li>python39Packages.openapi-core</li>
<li>python39Packages.openapi-schema-validator</li>
<li>python39Packages.openapi-spec-validator</li>
<li>python39Packages.openerz-api</li>
<li>python39Packages.papermill</li>
<li>python39Packages.phik</li>
<li>python39Packages.pika</li>
<li>python39Packages.pika-pool</li>
<li>python39Packages.plotly</li>
<li>python39Packages.pook</li>
<li>python39Packages.prance</li>
<li>python39Packages.praw</li>
<li>python39Packages.prawcore</li>
<li>python39Packages.pyarrow</li>
<li>python39Packages.pyee</li>
<li>python39Packages.pykwalify</li>
<li>python39Packages.pymatgen</li>
<li>python39Packages.pymodbus</li>
<li>python39Packages.pyppeteer</li>
<li>python39Packages.python-telegram-bot</li>
<li>python39Packages.pyutil</li>
<li>python39Packages.pywbem</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.qreactor</li>
<li>python39Packages.rachiopy</li>
<li>python39Packages.robomachine</li>
<li>python39Packages.robotframework</li>
<li>python39Packages.robotframework-databaselibrary</li>
<li>python39Packages.robotframework-requests</li>
<li>python39Packages.robotframework-sshlibrary</li>
<li>python39Packages.robotstatuschecker</li>
<li>python39Packages.robotsuite</li>
<li>python39Packages.scikit-tda</li>
<li>python39Packages.scrapy</li>
<li>python39Packages.scrapy-deltafetch</li>
<li>python39Packages.scrapy-fake-useragent</li>
<li>python39Packages.scrapy-splash</li>
<li>python39Packages.setuptoolsTrial</li>
<li>python39Packages.slack-sdk</li>
<li>python39Packages.smart-open</li>
<li>python39Packages.snowflake-connector-python</li>
<li>python39Packages.snowflake-sqlalchemy</li>
<li>python39Packages.sopel</li>
<li>python39Packages.sphinxcontrib-openapi</li>
<li>python39Packages.structlog</li>
<li>python39Packages.stytra</li>
<li>python39Packages.sumo</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.swagger-spec-validator</li>
<li>python39Packages.synergy</li>
<li>python39Packages.tensorboardx</li>
<li>python39Packages.testfixtures</li>
<li>python39Packages.towncrier</li>
<li>python39Packages.treq</li>
<li>ttp (python39Packages.ttp)</li>
<li>python39Packages.tubes</li>
<li>python39Packages.twisted</li>
<li>python39Packages.txaio</li>
<li>python39Packages.txamqp</li>
<li>python39Packages.txdbus</li>
<li>python39Packages.txgithub</li>
<li>python39Packages.txredisapi</li>
<li>python39Packages.txrequests</li>
<li>python39Packages.txtorcon</li>
<li>python39Packages.vega</li>
<li>vncdo (python39Packages.vncdo)</li>
<li>python39Packages.warlock</li>
<li>python39Packages.webthing</li>
<li>python39Packages.west</li>
<li>python39Packages.widgetsnbextension</li>
<li>python39Packages.zetup</li>
<li>python39Packages.zfec</li>
<li>python39Packages.zigpy-znp</li>
<li>qmk</li>
<li>rmview</li>
<li>snakemake</li>
<li>sourcehut.buildsrht</li>
<li>sourcehut.coresrht</li>
<li>sourcehut.dispatchsrht</li>
<li>sourcehut.gitsrht</li>
<li>sourcehut.hgsrht</li>
<li>sourcehut.hubsrht</li>
<li>sourcehut.listssrht</li>
<li>sourcehut.mansrht</li>
<li>sourcehut.metasrht</li>
<li>sourcehut.pastesrht</li>
<li>sourcehut.todosrht</li>
<li>streamlit</li>
<li>syncplay</li>
<li>tahoe-lafs</li>
<li>thrift</li>
<li>tinyprog</li>
<li>tts</li>
<li>urlwatch</li>
<li>vorta</li>
<li>weechatScripts.weechat-matrix</li>
<li>whipper</li>
</ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/tubes/default.nix:
```diff
@@ -0,0 +1,29 @@
+  src = fetchPypi {
+    pname = "Tubes";
+    inherit version;
+    sha256 = "sha256:0sg1gg2002h1xsgxigznr1zk1skwmhss72dzk6iysb9k9kdgymcd";
```
```suggestion
    sha256 = "0sg1gg2002h1xsgxigznr1zk1skwmhss72dzk6iysb9k9kdgymcd";
```

SuperSandro2000 commented on pkgs/development/python-modules/txtorcon/default.nix:
```diff
@@ -15,21 +16,19 @@ buildPythonPackage rec {
-  # https://github.com/meejah/txtorcon/issues/250
   checkPhase = ''
-    pytest --ignore=test/test_util.py .
+    python -m twisted.trial -j $NIX_BUILD_CORES ./test
```
```suggestion
    ${python.interpreter} -m twisted.trial -j $NIX_BUILD_CORES ./test
```

SuperSandro2000 commented on pkgs/development/python-modules/txtorcon/default.nix:
```diff
@@ -15,21 +16,19 @@ buildPythonPackage rec {
     homepage = "https://github.com/meejah/txtorcon";
-    maintainers = with lib.maintainers; [ jluttine ];
+    maintainers = with lib.maintainers; [ jluttine exarkun ];
     # Currently broken on Python 2.7. See
     # https://github.com/NixOS/nixpkgs/issues/71826
     broken = isPy27;
```
This can probably be moved to an disabled olderThan X.

SuperSandro2000 commented on pkgs/development/python-modules/txtorcon/default.nix:
```diff
@@ -1,12 +1,13 @@
 , GeoIP, isPy27}:
 
-buildPythonPackage rec {
+python3Packages.buildPythonPackage rec {
```
```suggestion
buildPythonPackage rec {
```

SuperSandro2000 commented on pkgs/development/python-modules/txtorcon/default.nix:
```diff
@@ -1,12 +1,13 @@
-{lib, buildPythonPackage, fetchPypi, isPy3k, incremental, ipaddress, twisted
+{ lib, python3Packages, fetchpatch, fetchPypi, isPy3k, incremental, ipaddress, twisted
```
```suggestion
{ lib, buildPythonPackage, pythonOlder, fetchpatch, fetchPypi, isPy3k, incremental, ipaddress, twisted
```

SuperSandro2000 commented on pkgs/development/python-modules/txtorcon/default.nix:
```diff
@@ -15,24 +16,21 @@ buildPythonPackage rec {
-  # https://github.com/meejah/txtorcon/issues/250
+  # Based on what txtorcon tox.ini will automatically test, allow back as far
+  # as Python 3.5.
+  disabled = python3Packages.pythonOlder "3.5"
```
```suggestion
  disabled = pythonOlder "3.5"
```

SuperSandro2000 reviewed (changes requested):
I just noticed that we can't use python3Packages as inputs anywhere in python-modules.

SuperSandro2000 commented:
@exarkun please resolve the merge conflict...

exarkun commented:
> @exarkun please resolve the merge conflict...

:disappointed:
